PEP 622: Add exception semantics

dmoisset opened the pull request:
Add a paragraph clarifying exception semantics as described in gvanrossum/patma#111

dmoisset committed:
Add exception semantics

gvanrossum commented on pep-0622.rst:
```diff
@@ -686,6 +686,16 @@ Finally, all attributes are exposed for matching, if a class wants to hide
+-------------------
+
+While matching each case, the ``match`` statement may trigger execution of other
+functions (for example ``__getitem__()``, ``__len__()`` or ``__get__()`` of some
+descriptor). Almost every exception caused by those propagates outside of the
```
Honestly instead of "`__get__` of some descriptor" I'd just use "a property" -- almost everybody knows properties, while their generalization `__get__` feels pretty esoteric.
```suggestion
functions (for example ``__getitem__()``, ``__len__()`` or
a property). Almost every exception caused by those propagates outside of the
```

gvanrossum reviewed (commented):
One small nit and then you can submit a PR against the python/peps repo.

dmoisset committed:
Give a better example

@gvanrossum suggested: «almost everybody knows properties, while their generalization __get__ feels pretty esoteric.»

Co-authored-by: Guido van Rossum <guido@python.org>

dmoisset commented:
Submitted this as PR1491 in python/peps. Closing
